pkg/assets/builder.go

```diff
@@ -113,7 +127,33 @@ func (a *AssetBuilder) RemapImage(image string) (string, error) {
 		image = asset.DockerImage
 	}
 
-	a.Assets = append(a.Assets, asset)
+	a.ContainerAssets = append(a.ContainerAssets, asset)
 
 	return image, nil
 }
+
+// RemapFile sets a new url location for the file, if a AssetsLocation is defined.
+func (a AssetBuilder) RemapFile(file string) (string, error) {
+	if file == "" {
+		return "", fmt.Errorf("unable to remap an empty string")
+	}
+
+	fileAsset := &FileAsset{
+		File:              file,
+		CanonicalLocation: file,
+	}
+
+	if a.AssetsLocation != nil && a.AssetsLocation.FileRepository != nil {
+		fileURL, err := url.Parse(file)
+		if err != nil {
+			return "", fmt.Errorf("unable to parse file url %q: %v", file, err)
+		}
+
+		fileRepo := strings.TrimSuffix(*a.AssetsLocation.FileRepository, "/")
+		fileAsset.File = fileRepo + fileURL.Path
```

Comment:
It may turn out we want the Host in here also, as otherwise we risk collisions (and it'll be more self explanation with the Host I would guess), but we can see how it goes...